### PR TITLE
separated string hashing related logic from OpBase to another class (HashableClass) to make OpBase simpler and more newcomer friendly

### PR DESCRIPTION
**Pull request by YoelShoshan**
* separated string hashing related logic from OpBase to another class (HashableClass) to make OpBase simpler and more newcomer friendly

* added unit test-reports and png files in main dir to .gitignore

### CODE REVIEW

File: fuse/data/ops/hashable_class.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+from fuse.data.ops import get_function_call_str
+from inspect import stack
+from fuse.data.ops.caching_tools import get_callers_string_description, value_to_string
+
+class HashableClass:
+    _MISSING_SUPER_INIT_ERR_MSG = 'Did you forget to call super().__init__() You must call it in your __init__? Also, make sure you call it BEFORE setting any attribute.'
+    
+    #controls how values are converted to string, override if you want custom behavior
+    #expected signature: foo(val:Any) -> str
+    value_to_string_func: Callable = value_to_string
```
Comment (mosheraboh) [inline fuse/data/ops/hashable_class.py:11]: Can you change to upper case?

File: fuse/data/ops/hashable_class.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
+from typing import Dict, Union, List, Sequence, Any, Optional, Callable
+from fuse.data.ops import get_function_call_str
+from inspect import stack
+from fuse.data.ops.caching_tools import get_callers_string_description, value_to_string
+
+class HashableClass:
```
Comment (mosheraboh) [inline fuse/data/ops/hashable_class.py:6]: Theoretically, we can move HashableClass, get_callers_string_description, and value_to_string to utils and to use it in other cases as well.
Leave it to you to decide.

File: fuse/data/ops/hashable_class.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        super().__setattr__(name, value)
+
+    
+    def get_hashable_string_representation(self) -> str:
+        '''
+        A string representation of this operation, which will be used for hashing.
+        It includes recorded (string) data describing the args that were used in __init__()
+        you can override/extend it in the rare cases that it's needed.
+
+        Note - not using __str__ or __repr__ by design, to avoid cases that developers override their Ops without being aware of the effect on caching
```
Comment (mosheraboh) [inline fuse/data/ops/hashable_class.py:48]: Need to update the comment here.

**Review by mosheraboh — APPROVED** [review]
Thanks!
Minors inline.

File: fuse/data/ops/hashable_class.py (hunk 3 above)
Comment (YoelShoshan) [reply fuse/data/ops/hashable_class.py:48]: can you please elaborate?

Comment (mosheraboh) [reply fuse/data/ops/hashable_class.py:45]: My bad, I saw you've mentioned __str__ method and didn't read it.
Unrealted, you have test failures.

Comment (YoelShoshan) [reply fuse/data/ops/hashable_class.py:45]: Thanks! Forgot to add another commit, added it now, the test says "Pending — This commit is scheduled to be built" on github, we should consider allocating more resources to it (and/or shortening the unit tests if possible)

**Review by mosheraboh — APPROVED** [review]

**Pull request merged**